packages/client/hmi-client/src/components/model/tera-parameters-metadata.vue

```diff
@@ -145,76 +42,45 @@ const emit = defineEmits(['update-parameter']);
 const parameters = computed(() => getParameters(props.model));
 const collapsedParameters = computed(() => collapseParameters(props.mmt, props.mmtParams));
 const parameterList = computed<
-	{ baseParameter: ModelParameter; childParameters: ModelParameter[]; isVirtual: boolean }[]
+	{
+		base: ModelVariable;
+		children: ModelVariable[];
+		isParent: boolean;
+	}[]
 >(() =>
 	Array.from(collapsedParameters.value.keys())
 		.flat()
 		.map((id) => {
 			const childIds = collapsedParameters.value.get(id) ?? [];
-			const childParameters = childIds
-				.map((childId) => parameters.value.find((p) => p.id === childId))
-				.filter(Boolean) as ModelParameter[];
-			const isVirtual = childIds.length > 1;
+			const isParent = childIds.length > 1;
+			const children = childIds
+				.map((childId) => {
+					const p = parameters.value.find((param) => param.id === childId);
+					// console.log(p, p?.units?.expression ?? '');
+					return {
+						id: p?.id ?? '',
+						name: p?.name ?? '',
+						description: p?.description ?? '',
+						grounding: p?.grounding,
+						unitExpression: p?.units?.expression ?? ''
+					};
+				})
+				.filter(Boolean) as ModelVariable[];
 
-			// If the parameter is virtual, we need to get the parameter data from model.metadata
-			const baseParameter = isVirtual
-				? props.model.metadata?.parameters?.[id] ?? { id } // If we haven't saved it in the metadata yet, create it
-				: parameters.value.find((p) => p.id === id);
+			const baseParameter = isParent ? { id } : parameters.value.find((p) => p.id === id);
+			const base = isParent
+				? { id: baseParameter?.id ?? '' }
+				: {
+						id: baseParameter?.id ?? '',
+						name: baseParameter?.name ?? '',
+						description: baseParameter?.description ?? '',
+						grounding: baseParameter?.grounding,
+						unitExpression: baseParameter?.units?.expression ?? ''
+					};
```

Comment:
This kind of pattern is rather hard to read, is everything actually nullable or is this just because of typescript?

```
const blah = foobar.find(....)

return {
   a: blah?.a ?? '',
   b: blah?.b ?? ''
}
```

Reply:
Cleaned up the assignment here as well as in `tera-states-metadata`.